libr/bin/p/bin_mz.c

```diff
@@ -58,22 +58,20 @@ static bool check_bytes(const ut8 *buf, ut64 length) {
 		}
 
 		// Check for New Executable, LE/LX or Phar Lap executable
-		if (!memcmp (buf + new_exe_header_offset, "NE", 2) ||
-			!memcmp (buf + new_exe_header_offset, "LE", 2) ||
+		if (!memcmp(buf, "MZ", 2) && 
```

Comment:
But why do you check here for MZ again? It was checked before in this function already. This looks wrong to me.

Reply:
And where is LE check?

Reply:
LE check was left out because of one of the binary in the r2-regressions repo cdogs is an LE but it is in the mz bins folder. due to which the build was failing.as you said the mz check was unnecessary and it was bad code on my part.

Reply:
Are you sure it's an LE file? If yes then that means the test is wrong

Reply:
In my analysis using file command, it showed up as LE file, for further analysis I made the bin_mz plugin return false at the end (line no: 73) and removed all the conditions in the if statement having only the `!memcmp (buf + new_exe_header_offset, "LE", 2)` condition and returned true for the statement.when I tested these conditions on the cdogs plugin it still showed up as mz. I also tried using a plugin for le which uses `r_read_ble16 () ` function to find the offset .this also returned as le

Reply:
Hey you actually right it is an LE file. So the test is wrong.

Reply:
Can you add back a check for LE and send a Pull request on radare2-regressions side adding BROKEN=1 on the faulty tests?

Reply:
yeah I will clean up this code and create a pull request in r2 regressions repo